src/Bundle/Maker/MakeFactoryPHPDocMethod.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Zenstruck\Foundry\Bundle\Maker;
+
+final class MakeFactoryPHPDocMethod
+{
+    public function __construct(private string $prototype, private bool $returnsCollection, private bool $isStatic = true, private bool $isRepository = false)
+    {
+    }
+
+    public function render(string $objectName, bool $withPHPStanEnabled): string
+    {
+        $annotation = $withPHPStanEnabled ? 'phpstan-method' : 'method';
+        $static = $this->isStatic ? 'static' : '      ';
+
+        $proxyType = $this->isRepository ? 'RepositoryProxy' : 'Proxy';
```

Comment:
I'm pretty sure there is a problem with repository here... and I've just realized there is no test which covers a case with an entity with repository. I'm gonna fix this in this PR

Reply:
I'm not following.

Reply:
we're not testing to create a factory for an entity which uses a repository, and my implementation has a problem, I've figured out when reviewing

Reply:
Ah ok, got it